File: lec1/examples/linux-c/fork.c

// fork.c: create a new process
#include<stdio.h>
#include <unistd.h>
int main()
{
    int pid;
    pid = fork();
    printf("fork() returned %d\n", pid);
    if (pid == 0)
    {
        printf("child\n");
    }
    else
    {
        printf("parent\n");
    }
}

File: lec1/examples/linux-rust/src/fork.rs
use fork::{fork, Fork};
fn main() {
    // keeps the same arguments of course
    match fork() {
       Ok(Fork::Parent(child)) => println!("Continuing execution in parent process, new child has pid: {}", child),
       Ok(Fork::Child) => println!("I'm a new child process"),
       Err(_) => println!("Fork failed"),
    }
}